File: PYTHON/Ejercicio22.py
#.Hallar: S = x^2/2! – x^4/4! + x8/6! – x^16/8! +…
import math
N=int(input("Intro N: "))
x=int(input("Intro x: "))
S=0; e=2; d=2
for i in range (1,N+1):
    if i%2==0:
        print("(",x,"^",e,") /",d,"! +")
        S=S-(x**e)/math.factorial(d)
    else:
        print("(",x,"^",e,") /",d,"! -")
        S=S+(x**e)/math.factorial(d)
    e=e*2
    d=d+2
print(" = ",S)

File: JAVA/Ejercicio22.java
import java.util.Scanner;
public class Ejercicio22 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Intro N: ");
        int N = sc.nextInt();
        System.out.print("Intro x: ");
        int x = sc.nextInt();
        double S = 0;
        int e = 2;
        int d = 2;
        for (int i = 1; i <= N; i++) {
            if (i % 2 == 0) {
                System.out.println("(" + x + "^" + e + ") / " + d + "! + ");
                S = S - (Math.pow(x, e) / factorial(d));
            } else {
                System.out.println("(" + x + "^" + e + ") / " + d + "! - ");
                S = S + (Math.pow(x, e) / factorial(d));
            }
            e = e * 2;
            d = d + 2;
        }
        System.out.println(" = " + S);
    }
    public static int factorial(int num) {
        int fact = 1;
        for (int i = 1; i <= num; i++) {
            fact *= i;
        }
        return fact;
    }
}
